pkg/resources/view_grant.go

```diff
@@ -107,14 +108,24 @@ func CreateViewGrant(data *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	// ID format is <db_name>|<schema_name>|<view_name>|<privilege>
-	// view_name is empty when on_future = true
-	if futureViews {
-		data.SetId(fmt.Sprintf("%v|%v||%v", dbName, schemaName, priv))
-	} else {
-		data.SetId(fmt.Sprintf("%v|%v|%v|%v", dbName, schemaName, viewName, priv))
+	dataIdentifiers := make([][]string, 1)
```

Comment:
we got rid of a conditional that says  if `futureViews ==true` then `viewName` must be an empty string

Reply:
how it was coded before it would look like 
```
if futureViews {
  viewName = ""
}
```

Reply:
Oh, so we're supposed to include that conditional?

Reply:
but you can also just assert that 
```
if futureViews && (viewName != "") {
    // error out since this looks like an error case
}
```

Reply:
Can I add that line to the existing futureViews conditional that defines the builder?

```
	var builder snowflake.GrantBuilder
	if futureViews {
		builder = snowflake.FutureViewGrant(dbName, schemaName)
                viewName = ""
	} else {
		builder = snowflake.ViewGrant(dbName, schemaName, viewName)
	}
```

Reply:
I might do 
```
if futureViews && (viewName != "") {
    // error out since this looks like an error case
}
```
all the way at the top where you are doing validation, before any of the logic/processing

Reply:
since it is an invariant you want to hold throughout your function